connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -763,6 +789,30 @@ private boolean startTask(ConnectorTaskId taskId) {
         );
     }
 
+    private Callable<Void> getTaskCallable(final ConnectorTaskId taskId, final TaskStatus.State targetState) {
+        return new Callable<Void>() {
+            @Override
+            public Void call() throws Exception {
+                switch (targetState) {
+                    case RUNNING:
+                        try {
+                            startTask(taskId);
```

Comment:
In principle it would feel more natural, I agree.

But what concerns me is how we bundle all the startup and shutdown calls to the worker, in order to have the bulk execution and how the herder is notified that startup/shutdown has completed. Individual calls of `worker.startConnector(...)/startTask(...)` in a for loop wouldn't work, unless we shutdown the executor in every phase (in order to call `invokeAll` we need to have prepared all the tasks that are about to run in the herder. Which is kind of what we do now).

Question: How about we retire the `executor` in the Worker. It's only used for starting tasks.